service.go

```diff
@@ -0,0 +1,337 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	apiError "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+)
+
+// Struct to build service watcher which looks after target services.
+type ServiceWatcher struct {
+	ctx       Context
+	svcFilter labels.Selector
+	informer  cache.SharedInformer
+	// Name space where to install service watcher,
+	namespace string
+}
+
+// Build new service watcher
+func NewServiceWatcher(ctx Context, ns *string) ServiceWatcher {
+	svc := &ServiceWatcher{
+		ctx: ctx,
+	}
+	svc.svcFilter = svc.createServiceFilter()
+	svc.informer = svc.createSharedInformer()
+	svc.namespace = *ns
+	return *svc
+}
+
+func (svc *ServiceWatcher) checkIfNameSpaceExists() {
```

Comment:
```suggestion
func (svc *ServiceWatcher) checkNamespaceExists() {
```

maybe redundant to say if 🤷🏻

Reply:
+1